packages/components/ng-log-to-customer/src/controller.js

```diff
@@ -0,0 +1,168 @@
+import flatten from 'lodash/flatten';
+import uniqBy from 'lodash/uniqBy.js';
+import { DISPLAY_RULES } from './constants';
+
+export default class LogToCustomerCtrl {
+  /* @ngInject */
+  constructor(
+    $translate,
+    LogToCustomer,
+    $interval,
+    $timeout,
+    $sce,
+    $transclude,
+  ) {
+    this.$translate = $translate;
+    this.LogToCustomer = LogToCustomer;
+    this.$interval = $interval;
+    this.$timeout = $timeout;
+    this.$sce = $sce;
+    this.$transclude = $transclude;
+  }
+
+  $onInit() {
+    this.logs = [];
+    this.interval = null;
+    this.searchText = null;
+    this.glued = true;
+    this.url = null;
+    this.DISPLAY_RULES = DISPLAY_RULES;
+    this.errorMessage = null;
+    this.isRightTileSlotFilled = this.$transclude.isSlotFilled('tile');
+    this.fullScreen = !this.isRightTileSlotFilled;
+
+    this.keys = Object.keys(this.logKeys);
+    this.setUrlTimeout = null;
+
+    this.setUrlSource().then(() => this.startLog());
```

Comment:
`.then()` will be called even though you have an error as you catch it. You should avoid starting log if you don't have any `url`